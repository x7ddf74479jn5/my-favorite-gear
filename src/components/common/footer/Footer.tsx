import Link from "@material-ui/core/Link";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import type { FC } from "react";
import React from "react";

const useStyles = makeStyles((theme) => {
  return {
    footer: {
      padding: theme.spacing(2),
      marginTop: "auto",
    },
  };
});
const Footer: FC = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <Typography variant="body1" color="textSecondary" align="center">
        {"俺のオタクソング8選"}
      </Typography>
      <Typography variant="body2" color="textSecondary" align="center">
        {"©2020  "}
        <Link color="inherit" href="https://twitter.com/ptrst102">
          {"@ptrst102"}
        </Link>
      </Typography>
    </footer>
  );
};

export default Footer;
