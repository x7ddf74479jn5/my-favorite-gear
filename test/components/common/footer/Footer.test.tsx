import Footer from "components/common/footer/Footer";
import React from "react";

import { render } from "../../../test-utils";

describe("Footer", () => {
  it("matches snapshot", () => {
    const renderResult = render(<Footer />);
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const renderResult = render(<Footer />);

    expect(renderResult.getByRole("contentinfo")).toHaveAttribute(
      "class",
      expect.stringContaining("footer")
    );
    expect(renderResult.container).toHaveTextContent("My Favorite Gear");
    const linkElement = renderResult.getByRole("link");
    expect(linkElement).toHaveTextContent("@pandashark6");
    expect(linkElement).toHaveAttribute(
      "href",
      "https://twitter.com/pandashark6"
    );
  });
});
