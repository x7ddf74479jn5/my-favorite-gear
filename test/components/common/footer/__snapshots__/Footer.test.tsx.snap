// Vitest Snapshot v1

exports[`Footer > matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="css-1f9yhiv"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-y7759j-MuiTypography-root"
    >
      My Favorite Gear
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-9dbpsu-MuiTypography-root"
    >
      ©2021  
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
        href="https://twitter.com/pandashark6"
      >
        @pandashark6
      </a>
    </p>
  </footer>
</DocumentFragment>
`;

exports[`Footer matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="makeStyles-footer-1"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      My Favorite Gear
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      ©2021  
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://twitter.com/pandashark6"
      >
        @pandashark6
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
